### PR TITLE
Enable server side cursors

### LINKED ISSUE #35689: ENH: Support PostgreSQL server-side cursors to prevent memory hog on large datasets
#### Is your feature request related to a problem?

`pandas.read_sql_query` supports Python "generator" pattern when providing `chunksize` argument. It's not very helpful when working with large datasets, since the whole data is initially retrieved from DB into client-side memory and later chunked into separate frames based on `chunksize`. Large datasets will easily run into out-of-memory problems with this approach.

#### Describe the solution you'd like

Postgres/psycopg2 are addressing this problem with [server-side cursors](https://www.psycopg.org/docs/usage.html#server-side-cursors). But Pandas does not support it.

#### API breaking implications

`is_cursor` argument of `SQLDatabase` or `SQLiteDatabase` is not exposed in `pandas.read_sql_query` or `pandas.read_sql_table` without any reason. It should be exposed, this way named (server-side) cursor could be provided.

#### Describe alternatives you've considered

Instead of doing:

```
iter = sql.read_sql_query(sql,
      conn,
      index_col='col1',
      chunksize=chunksize)
```

I tried reimplementing it like this:

```
from pandas.io.sql import SQLiteDatabase

curs = conn.cursor(name='cur_name') # server side cursor creation
curs.itersize = chunksize

pandas_sql = SQLiteDatabase(curs, is_cursor=True)
iter = pandas_sql.read_query(
      sql,
      index_col='col1',
      chunksize=chunksize)
```

but it fails because `SQLiteDatabase` tries to access `cursor.description`, which is NULL for some reason with server-side cursors (and idea why?).

### PR DESCRIPTION
**Pull request by J0**
The old PR went stale and I ended up busy for a year so had to re-open in a new branch. Hoping to finally push this through

This pull request attempts to fix #35689.  I read an article by @itamarst and decided to look further into the codebase. After noticing that the fix semed to be simple I decided file a PR. The fix is naive and I'm not sure the true solution is as simple as I imagine it to be. There are probably many things I'm missing as I've not read the codebase in detail so do let me know what else needs to be addressed.

- [x] closes #35689
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry(N.A.)

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -1382,9 +1382,23 @@ def run_transaction(self):
         else:
             yield self.connectable
 
-    def execute(self, *args, **kwargs):
-        """Simple passthrough to SQLAlchemy connectable"""
-        return self.connectable.execution_options().execute(*args, **kwargs)
+    def execute(self, chunksize: int = 0, *args, **kwargs):
```
Comment (jreback) [inline pandas/io/sql.py:1385]: can you add a return type

**Review by jreback — CHANGES_REQUESTED** [review]
is this exercised in *any* current tests? 

does the doc-string need updating here?

File: pandas/io/sql.py (hunk 1 above)
Comment (J0) [reply pandas/io/sql.py:1385]: Will do!

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge the main branch, address the review comments, and we can reopen.

**Comment by gregorywaynepower** [comment]
@J0 Is this still being worked on?

**Pull request closed without merging**